src/main/java/com/yahoo/sketches/tuple/ArrayOfDoublesUnion.java

```diff
@@ -190,16 +190,23 @@ void setThetaLong(final long theta) {
   static ArrayOfDoublesUnion wrapUnionImpl(final WritableMemory mem, final long seed,
       final boolean isWritable) {
     final SerializerDeserializer.SketchType type = SerializerDeserializer.getSketchType(mem);
+    final ArrayOfDoublesQuickSelectSketch sketch;
+    final ArrayOfDoublesUnion union;
 
     // compatibility with version 0.9.1 and lower
     if (type == SerializerDeserializer.SketchType.ArrayOfDoublesQuickSelectSketch) {
-      final ArrayOfDoublesQuickSelectSketch sketch = isWritable
-          ? new DirectArrayOfDoublesQuickSelectSketch(mem, seed)
-              : new DirectArrayOfDoublesQuickSelectSketchR(mem, seed);
-      return isWritable ? new DirectArrayOfDoublesUnion(sketch, mem)
-          : new DirectArrayOfDoublesUnionR(sketch, mem);
+      if (isWritable) {
+        sketch = new DirectArrayOfDoublesQuickSelectSketch(mem, seed);
+        union = new DirectArrayOfDoublesUnion(sketch, mem);
+      } else {
+        sketch = new DirectArrayOfDoublesQuickSelectSketchR(mem, seed);
+        union = new DirectArrayOfDoublesUnionR(sketch, mem);
+      }
+      return union; //TODO set union theta?
```

Comment:
I notice that you do not set union.theta_ here as you do on line 232.  Oversight?